src/js/elements/point_tray/category.js

```diff
@@ -18,10 +18,10 @@ export class Category extends Component {
     fixData(data) {
         data = {...data}
 
-        if (data.metadata === undefined)
+        if (data.metadata === undefined || data.metadata === null)
             data.metadata = {}
 
-        if (data.metadata.source === undefined)
+        if (data.metadata?.source === undefined)
```

Comment:
Is the question mark supposed to be there?

Reply:
yup `?.` is an operator in ES6
if metadata is null or undefined it won't throw error if it didn't find source.. 
It will be just undefined
added it as a precaution